commercelayer/resource_adyen_gateway.go

```diff
@@ -0,0 +1,204 @@
+						},
+						"live_url_prefix": {
+							Description: "The prefix of the endpoint used for live transactions.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"reference": {
+							Description: "A string that you can use to add any external identifier to the resource. This " +
+								"can be useful for integrating the resource to an external system, like an ERP, a " +
+								"marketing tool, a CRM, or whatever.",
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"reference_origin": {
+							Description: "Any identifier of the third party system that defines the reference code",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+						"metadata": {
+							Description: "Set of key-value pairs that you can attach to the resource. This can be useful " +
+								"for storing additional information about the resource in a structured format",
+							Type: schema.TypeMap,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+							Optional: true,
+						},
+						"relationships": {
+							Description: "Resource relationships",
+							Type:        schema.TypeList,
+							MaxItems:    1,
+							MinItems:    1,
+							Optional:    true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"adyen_payment_id": {
+										Description: "The associated payment id.",
+										Type:        schema.TypeString,
+										Optional:    true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAdyenGatewayReadFunc(ctx context.Context, d *schema.ResourceData, i interface{}) diag.Diagnostics {
+	c := i.(*commercelayer.APIClient)
+
+	resp, _, err := c.AdyenGatewaysApi.GETAdyenGatewaysAdyenGatewayId(ctx, d.Id()).Execute()
+	if err != nil {
+		return diagErr(err)
+	}
+
+	adyenGateway, ok := resp.GetDataOk()
+	if !ok {
+		d.SetId("")
+		return nil
+	}
+
+	d.SetId(adyenGateway.GetId())
+
+	return nil
+}
+
+func resourceAdyenGatewayCreateFunc(ctx context.Context, d *schema.ResourceData, i interface{}) diag.Diagnostics {
+	c := i.(*commercelayer.APIClient)
+
+	attributes := nestedMap(d.Get("attributes"))
+
+	// TODO: Relationships when paymentMethods are also created
+	adyenGatewayCreate := commercelayer.AdyenGatewayCreate{
+		Data: commercelayer.AdyenGatewayCreateData{
+			Type: adyenGatewaysType,
+			Attributes: commercelayer.POSTAdyenGateways201ResponseDataAttributes{
+				Name:            attributes["name"].(string),
+				MerchantAccount: attributes["merchant_account"].(string),
+				ApiKey:          attributes["api_key"].(string),
+				LiveUrlPrefix:   attributes["live_url_prefix"].(string),
+				Reference:       stringRef(attributes["reference"]),
+				ReferenceOrigin: stringRef(attributes["reference_origin"]),
+				Metadata:        keyValueRef(attributes["metadata"]),
+			},
+			//	Relationships: &commercelayer.AdyenGatewayCreateDataRelationships{
```

Comment:
Can you check and remove any commented out code? If we don't use it then delete it!